src/Pipeline/Queue/Atomic.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) Daniele Alessandri <suppakilla@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Pipeline\Queue;
+
+use SplQueue;
+use Throwable;
+use Traversable;
+use NoRewindIterator;
+use InvalidArgumentException;
+use Predis\Command\CommandInterface;
+use Predis\Command\RawCommand;
+use Predis\Connection\ConnectionInterface;
+use Predis\Response\ErrorInterface;
+
+/**
+ * Atomic command queue that wraps flushes into MULTI / EXEC transactions.
+ */
+class Atomic implements CommandQueueInterface
+{
+    private $discarded = false;
+    /** @var CommandQueueInterface */
+    private $innerQueue;
+    /** @var SplQueue */
+    private $txCommands;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct(CommandQueueInterface $queue)
+    {
+        $this->checkInnerQueue($queue);
+
+        $this->innerQueue = $queue;
+        $this->txCommands = new SplQueue;
+    }
+
+    /**
+     * Verifies if the inner command queue is compatibile with atomic queues.
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function checkInnerQueue(CommandQueueInterface $queue): void
```

Comment:
We might want to rename this method so it clearly indicates intent, maybe something like `checkInnerQueueCompatibility` or `verifyInnerQueueIsCompatible`

Reply:
I like `checkInnerQueueCompatibility`, will use that.

Reply:
Dope!